crates/sui-types/src/messages_checkpoint.rs

```diff
@@ -489,6 +501,137 @@ impl CheckpointContents {
     }
 }
 
+/// Same as CheckpointContents, but contains full contents of all Transactions and
+/// TransactionEffects associated with the checkpoint.
+// TODO: rename to CheckpointContents after CheckpointContents is renamed CheckpointDigests.
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct FullCheckpointContents {
+    transactions: Vec<ExecutionData>,
+    /// This field 'pins' user signatures for the checkpoint
+    /// The length of this vector is same as length of transactions vector
+    /// System transactions has empty signatures
+    user_signatures: Vec<Vec<GenericSignature>>,
+}
```

Comment:
I think this needs to be restructured to the following:
```
struct FullCheckpointContents {
    contents: CheckpointContents,
    transactions: Vec<(Transaction, TransactionEffects)>,
}
```

All we want the RPC to do is bundle the txns and effects with the already existing CheckpointContents struct. In fact for the actual RPC we may even want request flags to indicate if txns or effects should be included in the payload. As in the future we may not want effects to be transmitted either (even though we will always request effects today)